### DISCUSSION TITLE
Props naming convention

### DISCUSSION THREAD

**Discussion by dlastnoel** [inertiajs/inertia#1929] [Help]
Hi! I'm just wondering, what is the proper way/naming convention when sharing data via props, is camelCase better than snake_case?

```bash
return Inertia::render('App/Home/Index', [
    'success_message' => 'Well done!'
]);
```

```bash
return Inertia::render('App/Home/Index', [
    'successMessage' => 'Well done!'
]);
```

**Comment by reinink** [comment]
I always use camelCase for top-level props, as that's the convention generally used for props in client-side components. However, I typically use snake_case for nested properties, as that's typically how they come from the database:

```php
return Inertia::render('Profile', [
    'currentUser' => [
        'first_name' => Auth::user()->first_name,
        'last_name' => Auth::user()->last_name,
        'email' => Auth::user()->email,
    ]
]);
```

**Reply by dlastnoel** [reply]
That's a good one. Thanks!
